tree/dataframe/src/RDFHelpers.cxx

```diff
@@ -57,3 +60,124 @@ void ROOT::RDF::RunGraphs(std::vector<RResultHandle> handles)
    for (auto &h : uniqueLoops)
       run(h);
 }
+
+
+/// Compute a running mean of events/s.
+double ROOT::RDF::ProgressHelper::EvtPerSec() const {
+  if (fEventsPerSecondStatisticsIndex < fEventsPerSecondStatistics.size())
+    return std::accumulate(fEventsPerSecondStatistics.begin(), fEventsPerSecondStatistics.begin() + fEventsPerSecondStatisticsIndex, 0.) / fEventsPerSecondStatisticsIndex;
+  else
+    return std::accumulate(fEventsPerSecondStatistics.begin(), fEventsPerSecondStatistics.end(), 0.) / fEventsPerSecondStatistics.size();
+}
+
+/// Record current event counts and time stamp, populate evts/s statistics array.
+std::pair<std::size_t, std::chrono::seconds>
+ROOT::RDF::ProgressHelper::RecordEvtCountAndTime() {
+  using namespace std::chrono;
+
+  const auto currentEventCount = fProcessedEvents.load();
+  const auto eventsPerTimeInterval = currentEventCount - fLastProcessedEvents;
+  fLastProcessedEvents = currentEventCount;
+
+  const auto oldPrintTime = fLastPrintTime;
+  fLastPrintTime = system_clock::now();
+
+  const duration<double> secondsCurrentInterval = fLastPrintTime - oldPrintTime;
+  fEventsPerSecondStatistics[fEventsPerSecondStatisticsIndex++ % fEventsPerSecondStatistics.size()] = eventsPerTimeInterval / secondsCurrentInterval.count();
+
+  return {currentEventCount, duration_cast<seconds>(fLastPrintTime - fBeginTime)};
+}
+
+namespace {
+  struct PacksOfThree {
+    std::array<unsigned int, 7> packsOfThree; // More not supported by std::size_t
+    int packCounter = 0;
+
+    PacksOfThree(std::size_t count) {
+      for (; count > 0; ++packCounter) {
+        assert(packCounter < static_cast<int>(packsOfThree.size()));
+        packsOfThree[packCounter] = count % 1000;
+        count /= 1000;
+      }
+      --packCounter;
+    }
+  };
+
+  /// Format event counts as `6.346.362k`.
+  std::ostream& operator<<(std::ostream& stream, const PacksOfThree& packs) {
+    for (int i = static_cast<int>(packs.packCounter); i >= 1; --i) {
+      if (i == packs.packCounter) stream << packs.packsOfThree[i];
+      else stream << std::setw(3) << std::setfill('0') << std::right << packs.packsOfThree[i];
+
+      stream << (i > 1 ? '.' : 'k');
+    }
+    return stream << std::setfill(' ');
+  }
+
+  /// Format std::chrono::seconds as `1:30m`.
+  std::ostream& operator<<(std::ostream& stream, std::chrono::seconds elapsedSeconds) {
+    stream << std::chrono::duration_cast<std::chrono::minutes>(elapsedSeconds).count()
+        << ":" << std::setw(2) << std::right << std::setfill('0') << elapsedSeconds.count() % 60;
+    return stream << 'm' << std::setfill(' ');
```

Comment:
`setfill(' ')` might not be resetting it to the *prior* state, and `setw` might need to be reset, too? See https://stackoverflow.com/questions/2273330/restore-the-state-of-stdcout-after-manipulating-it

Reply:
👏

Reply:
@Axel-Naumann `fmtflags` is not saving the fill character, is it?

Reply:
I don't think so, but I fixed that.